### ISSUE TITLE
False positive report: DynamoDB table encryption

### ISSUE THREAD

**Issue by christophetd** [fugue/regula#198]
Hello,

Regula fails when a resource `aws_dynamodb_table` does not specify the [`server_side_encryption`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/dynamodb_table#server_side_encryption) parameter. 

```
FG_R00069: DynamoDB tables should be encrypted with AWS or customer managed KMS CMKs [Medium]

  [1]: xxx.my_dynamo_db
       in ./path/to/main.tf:1:1
       included at main.tf:2:30
```

However, quoting the docs:

> AWS DynamoDB tables are automatically encrypted at rest with an AWS owned Customer Master Key if this argument isn't specified.

So this looks like a false positive.

**Comment by jason-fugue** [comment]
Hi, @christophetd! It looks like we don't surface the full rule description in an effective way, but the full description clarifies this a little: https://github.com/fugue/regula/blob/master/rego/rules/tf/aws/dynamodb/dynamodb_encryption.rego#L25

> Although DynamoDB tables are encrypted at rest by default with AWS owned CMKs, using AWS managed CMKs or customer managed CMKs provides additional functionality via AWS KMS, such as viewing key policies, auditing usage, and rotating cryptographic material.

The distinction is between AWS owned vs AWS managed keys. The AWS docs have more information about what the differences are: https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/encryption.howitworks.html

**Comment by jaspervdj-luminal** [comment]
Hi @christophetd,

This sounded like a false positive to me at first as well, but after going through the docs again I'm not so sure.  There's [three types of KMS keys](https://docs.aws.amazon.com/kms/latest/developerguide/concepts.html#kms_keys):

- AWS Owned
- AWS Managed
- Customer Managed

DynamoDB is always encrypted using one of these three; there's no "none" setting.  However, AWS Owned keys are reused by AWS across different accounts, which is a reason to avoid them.  So the rule is saying:

- AWS Owned (fail)
- AWS Managed (pass)
- Customer Managed (pass)

However, I agree that the description is awkwardly worded and needs to make the distinction clear between "AWS Owned" and "AWS Managed".

**Comment by christophetd** [comment]
Very good point and very nice finding!  Shall we leave the issue open as an improvement rather than a bug, then?

**Comment by fugue-chris** [comment]
@christophetd yep - we will leave this open until we clarify things in the metadata and documentation. Thanks for raising this.

-Chris
